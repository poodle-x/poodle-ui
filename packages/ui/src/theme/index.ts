import React from "react";
import {
	ThemeValue,
	SystemThemeConfig,
	BreakpointsThemeConfig,
	SpacingThemeConfig,
	ColorsThemeConfig,
	FontsThemeConfig,
	BordersThemeConfig,
	ShadowThemeConfig,
	LayoutsThemeConfig,
} from "@poodle/system";

import { CSSObjectSystem, Stylesheet } from "../styled";
import { BoxThemeConfig } from "../Box";
import { ButtonThemeConfig } from "../Button";
import { ColumnsThemeConfig } from "../Columns";
import { ColumnThemeConfig } from "../Column";
import { ContainerThemeConfig } from "../Container";
import { TableThemeConfig } from "../Table";
import { TableHeadThemeConfig } from "../TableHead";
import { TableBodyThemeConfig } from "../TableBody";
import { TableRowThemeConfig } from "../TableRow";
import { TableCellThemeConfig } from "../TableCell";
import { VisuallyHiddenThemeConfig } from "../VisuallyHidden";
import { CheckControlThemeConfig } from "../CheckControl";
import { CheckboxThemeConfig } from "../Checkbox";
import { RadioThemeConfig } from "../Radio";
import { InputThemeConfig } from "../Input";
import { InputAdornmentThemeConfig } from "../InputAdornment";
import { IconButtonThemeConfig } from "../IconButton";
import { ModalThemeConfig } from "../Modal";
import { ModalHeaderThemeConfig } from "../ModalHeader";
import { ModalBodyThemeConfig } from "../ModalBody";
import { ModalFooterThemeConfig } from "../ModalFooter";
import { SelectThemeConfig } from "../Select";
import { TabsThemeConfig } from "../Tabs";
import { TabThemeConfig } from "../Tab";
import { TabListThemeConfig } from "../TabList";
import { TabPanelThemeConfig } from "../TabPanel";

export { Stylesheet };

export { useTheme } from "./useTheme";

export type DirectionThemeConfig = "rtl" | "ltr";

export interface GlobalThemeConfig {
	styles?: ThemeValue<CSSObjectSystem>;
}

export type IconType = React.FC<
	React.PropsWithoutRef<React.SVGProps<SVGSVGElement>> &
		React.RefAttributes<SVGSVGElement>
>;

export interface BaseIcons {
	RadioButton?: IconType;
	RadioCheckedButton?: IconType;
	CheckboxButton?: IconType;
	CheckboxCheckedButton?: IconType;
	CheckboxIndeterminateButton?: IconType;
	ArrowDown?: IconType;
}

export interface ThemeConfig extends SystemThemeConfig {
	global?: GlobalThemeConfig;
	direction?: ThemeValue<DirectionThemeConfig>;
	baseIcons?: BaseIcons;
	Box?: BoxThemeConfig;
	Button?: ButtonThemeConfig;
	Columns?: ColumnsThemeConfig;
	Column?: ColumnThemeConfig;
	Container?: ContainerThemeConfig;
	Table?: TableThemeConfig;
	TableHead?: TableHeadThemeConfig;
	TableBody?: TableBodyThemeConfig;
	TableRow?: TableRowThemeConfig;
	TableCell?: TableCellThemeConfig;
	VisuallyHidden?: VisuallyHiddenThemeConfig;
	CheckControl?: CheckControlThemeConfig;
	Checkbox?: CheckboxThemeConfig;
	Radio?: RadioThemeConfig;
	Input?: InputThemeConfig;
	InputAdornment?: InputAdornmentThemeConfig;
	IconButton?: IconButtonThemeConfig;
	Modal?: ModalThemeConfig;
	ModalHeader?: ModalHeaderThemeConfig;
	ModalBody?: ModalBodyThemeConfig;
	ModalFooter?: ModalFooterThemeConfig;
	Select?: SelectThemeConfig;
	Tabs?: TabsThemeConfig;
	Tab?: TabThemeConfig;
	TabList?: TabListThemeConfig;
	TabPanel?: TabPanelThemeConfig;
}

export const defaultGlobal = (
	overrideTheme: ThemeConfig
): GlobalThemeConfig => {
	return {
		styles: overrideTheme.global?.styles,
	};
};

export const defaultBreakpoints = (
	overrideTheme: ThemeConfig
): BreakpointsThemeConfig => {
	return {
		scale: ["sm", "md", "lg", "xl"],
		...overrideTheme.breakpoints,
		sets: {
			sm: 576,
			md: 768,
			lg: 992,
			xl: 1200,
			xxl: 1400,
			...overrideTheme.breakpoints?.sets,
		},
	};
};

export const defaultSpacing = (
	overrideTheme: ThemeConfig
): SpacingThemeConfig => {
	return {
		scale: 4,
		...overrideTheme.spacing,
		sets: {
			s: 4,
			m: 8,
			l: 16,
			...overrideTheme.spacing?.sets,
		},
	};
};

export const defaultFonts = (overrideTheme: ThemeConfig): FontsThemeConfig => {
	return {
		scale: "1rem",
		...overrideTheme.fonts,
		base: {
			fontFamily:
				'-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"',
			fontStyle: "normal",
			fontWeight: "normal",
			lineHeight: "125%",
			fontSize: "1rem",
			...overrideTheme.fonts?.base,
		},
		weights: {
			normal: 400,
			semibold: 600,
			bold: 700,
			...overrideTheme.fonts?.weights,
		},
		sets: {
			input: {
				fontSize: "1rem",
				fontWeight: "normal",
				lineHeight: "125%",
			},
			button: {
				fontSize: "1rem",
				fontWeight: "bold",
				lineHeight: "150%",
			},
			label: {
				fontStyle: "normal",
				fontWeight: "bold",
				fontSize: "0.875rem",
				lineHeight: "110%",
			},
			helper: {
				fontStyle: "normal",
				fontWeight: "normal",
				fontSize: "0.875rem",
				lineHeight: "110%",
			},
			checkControlLabelMedium: {
				fontStyle: "normal",
				fontWeight: "normal",
				fontSize: "1rem",
				lineHeight: "110%",
			},
			checkControlLabelSmall: {
				fontStyle: "normal",
				fontWeight: "normal",
				fontSize: "0.875rem",
				lineHeight: "110%",
			},
			checkControlLabelLarge: {
				fontStyle: "normal",
				fontWeight: "normal",
				fontSize: "1.125rem",
				lineHeight: "110%",
			},
			modalTitle: {
				fontStyle: "normal",
				fontWeight: "bold",
				fontSize: "1.5rem",
				lineHeight: "133.33%",
			},
			...overrideTheme.fonts?.sets,
		},
	};
};

export const defaultColor = (overrideTheme: ThemeConfig): ColorsThemeConfig => {
	return {
		modes: {
			...overrideTheme.colors?.modes,
		},
		sets: {
			text: "#282828",
			disabledText: "#A6A6A6",

			primary: {
				light: "#D1EBFF",
				base: "#00478A",
				dark: "#003A70",
				textOnLight: "#00478A",
				textOnBase: "#fff",
				textOnDark: "#fff",
			},

			negative: {
				base: "#AF1308",
				dark: "#8D1007",
				light: "#FCF3F3",
				textOnLight: "#AF1308",
				textOnBase: "#fff",
				textOnDark: "#fff",
			},

			positive: {
				base: "#006122",
				dark: "#004D1B",
				light: "#E5FFEE",
				textOnLight: "#006122",
				textOnBase: "#fff",
				textOnDark: "#fff",
			},

			warn: {
				base: "#FFBB33",
				dark: "#FAAB00",
				light: "#FFF1DB",
				textOnLight: "#282828",
				textOnBase: "#282828",
				textOnDark: "#282828",
			},

			bg: {
				base: "#F8F8F8",
				100: "#E0E0E0",
				textOn100: "#282828",
				200: "#CCCCCC",
				textOn200: "#282828",
			},

			modal: {
				base: "#F8F8F8",
				footer: "#F0F0F0",
			},

			border: {
				base: "#CCCCCC",
			},

			...overrideTheme.colors?.sets,
		},
	};
};

export const defaultBorder = (
	overrideTheme: ThemeConfig
): BordersThemeConfig => {
	return {
		sets: {
			base: {
				style: "solid",
				width: "1px",
				color: "border",
			},
			...overrideTheme?.borders?.sets,
		},
		radius: {
			base: "4px",
			...overrideTheme?.borders?.radius,
		},
	};
};

export const defaultBoxShadow = (
	overrideTheme: ThemeConfig
): ShadowThemeConfig => {
	return {
		boxShadowSets: {
			lv1: "1px 2px 2px -1px rgba(0,0,0,0.4)",
			lv2: "2px 2px 5px 0px rgba(0,0,0,0.4)",
			lv3: "4px 4px 5px 0px rgba(0,0,0,0.4)",
			...overrideTheme?.shadows?.boxShadowSets,
		},
	};
};

export const defaultLayouts = (
	overrideTheme: ThemeConfig
): LayoutsThemeConfig => {
	return {
		zIndices: {
			modal: 2000,
			...overrideTheme?.layouts?.zIndices,
		},
	};
};

export function createTheme(overrideTheme: ThemeConfig = {}): ThemeConfig {
	return {
		direction: "ltr",
		...overrideTheme,
		global: defaultGlobal(overrideTheme),
		spacing: defaultSpacing(overrideTheme),
		breakpoints: defaultBreakpoints(overrideTheme),
		colors: defaultColor(overrideTheme),
		fonts: defaultFonts(overrideTheme),
		borders: defaultBorder(overrideTheme),
		shadows: defaultBoxShadow(overrideTheme),
		layouts: defaultLayouts(overrideTheme),
	};
}

export type StylesConfig<T extends string> = {
	[K in T]: ThemeValue<CSSObjectSystem>;
};

export interface StandardComponentProps {
	variant?: string;
	themeExtend?: ThemeConfig;
	theme?: ThemeConfig;
	sizeStyle?: string;
	colorStyle?: string;
}

export interface StandardThemeConfig<
	T extends string,
	Props extends { [key: string]: any }
> {
	defaultProps?: Partial<Props>;
	styles?: Partial<StylesConfig<T>>;
	variants?: {
		[key: string]: Partial<StylesConfig<T>>;
	};
	sizes?: {
		[key: string]: Partial<StylesConfig<T>>;
	};
	colorStyles?: {
		[key: string]: Partial<StylesConfig<T>>;
	};
	modes?: {
		[key: string]: {
			styles?: Partial<StylesConfig<T>>;
			variants?: {
				[key: string]: Partial<StylesConfig<T>>;
			};
			sizes?: {
				[key: string]: Partial<StylesConfig<T>>;
			};
			colorStyles?: {
				[key: string]: Partial<StylesConfig<T>>;
			};
		};
	};
	overrides?: Partial<StylesConfig<T>>;
	overrideClasses?: Partial<
		{
			[K in T]: boolean | string;
		}
	>;
}

export * from "./system";
