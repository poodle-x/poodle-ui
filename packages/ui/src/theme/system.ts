import {
	BordersProps,
	bordersSystem,
	ColorsProps,
	colorsSystem,
	compose,
	createSystem,
	FlexProps,
	flexSystem,
	FontsProps,
	fontsSystem,
	getMode,
	GridProps,
	gridSystem,
	LayoutProps,
	layoutSystem,
	ListProps,
	listSystem,
	PseudosProps,
	pseudosSystem,
	ResponsiveProp,
	ShadowProps,
	shadowSystem,
	SpacingProps,
	spacingSystem,
	SystemConfigList,
} from "@poodle/system";
import * as CSS from "csstype";
import { CSSSystem } from "../styled";
import { StandardThemeConfig } from "./index";

const additionalSystem: SystemConfigList = {
	objectFit: true,
	objectPosition: true,
	transition: true,
	cursor: true,
	pointerEvents: true,
};

export interface AdditionalSystemProps {
	/**
	 * The CSS `object-fit` property
	 */
	objectFit?: ResponsiveProp<CSS.Property.ObjectFit>;
	/**
	 * The CSS `object-fit` property
	 */
	objectPosition?: ResponsiveProp<CSS.Property.ObjectPosition>;
	/**
	 * The CSS `transition` property
	 */
	transition?: ResponsiveProp<CSS.Property.Transition>;
	/**
	 * The CSS `cursor` property
	 */
	cursor?: ResponsiveProp<CSS.Property.Cursor>;
	/**
	 * The CSS `cursor` property
	 */
	pointerEvents?: ResponsiveProp<CSS.Property.PointerEvents>;
}

const allSystem = compose(
	colorsSystem,
	fontsSystem,
	layoutSystem,
	spacingSystem,
	bordersSystem,
	flexSystem,
	pseudosSystem,
	shadowSystem,
	gridSystem,
	listSystem,
	additionalSystem
);

export const system = createSystem(allSystem);

interface CssSystemProps
	extends SpacingProps,
		LayoutProps,
		FontsProps,
		ColorsProps,
		BordersProps,
		ShadowProps,
		FlexProps,
		GridProps,
		ListProps,
		AdditionalSystemProps {}

export interface SystemProps
	extends SpacingProps,
		LayoutProps,
		FontsProps,
		ColorsProps,
		BordersProps,
		ShadowProps,
		FlexProps,
		PseudosProps<CssSystemProps>,
		GridProps,
		ListProps,
		AdditionalSystemProps {}

const additionalProps = [
	"textStyle",
	"variant",
	"themeExtend",
	"theme",
	"sizeStyle",
	"colorStyle",
];

// Cache to avoid unnecessary repeat loop
// @todo maybe move to a helper func
const finalSystem: SystemConfigList = {};

for (const keys in allSystem) {
	const keysPart = keys.split(",");

	const rawConfig = allSystem[keys];

	keysPart.forEach((k) => {
		finalSystem[k] = rawConfig;
	});
}

export function removeSystemProps(props: { [key: string]: any }) {
	let resultProps = {};

	for (const prop in props) {
		let found = false;

		if (finalSystem[prop] || additionalProps.indexOf(prop) >= 0) {
			found = true;
		}

		if (!found && ["sx"].indexOf(prop) === -1) {
			resultProps = {
				...resultProps,
				[prop]: props[prop],
			};
		}
	}

	return resultProps;
}

export function createCSSSystemStandard<Key extends string>(data: {
	key: Key;
	config?: StandardThemeConfig<Key, any>;
	props: { [key: string]: any };
	base?: CSSSystem["base"];
}): CSSSystem {
	const { key, props, config, base } = data;

	const { variant = "", sizeStyle = "", colorStyle = "" } = props;

	const mode = getMode(props);

	return {
		base,
		applies: [
			config?.styles?.[key],
			config?.modes?.[mode]?.styles?.[key],
			config?.variants?.[variant]?.[key],
			config?.modes?.[mode]?.variants?.[variant]?.[key],
			config?.sizes?.[sizeStyle]?.[key],
			config?.modes?.[mode]?.sizes?.[sizeStyle]?.[key],
			config?.colorStyles?.[colorStyle]?.[key],
			config?.modes?.[mode]?.colorStyles?.[colorStyle]?.[key],
		],
		override: config?.overrides?.[key],
	};
}

export * from "@poodle/system";
