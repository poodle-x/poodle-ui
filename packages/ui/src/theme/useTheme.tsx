import { ThemeContext } from "@emotion/react";
import deepmerge from "deepmerge";
import React from "react";
import { ThemeConfig } from "./";

export function useTheme(
	options: {
		override?: ThemeConfig;
		extend?: ThemeConfig;
	} = {}
): { theme: ThemeConfig } {
	const { override, extend } = options;

	const themeContext: ThemeConfig = React.useContext(ThemeContext);

	const returnTheme = React.useMemo(() => {
		if (override) {
			return override;
		}

		if (extend) {
			return deepmerge(themeContext, extend, {
				arrayMerge: (_, sourceArray) => {
					return sourceArray;
				},
			});
		}

		return themeContext;
	}, [extend, override, themeContext]);

	return {
		theme: returnTheme || {},
	};
}
