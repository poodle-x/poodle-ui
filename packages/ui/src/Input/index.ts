import { Input, InputProps } from "./Input";
import { StandardThemeConfig } from "../theme";

export * from "./Input";

export default Input;

export type InputStyleKeys = "root" | "input";

export interface InputThemeConfig
	extends StandardThemeConfig<InputStyleKeys, InputProps> {}
