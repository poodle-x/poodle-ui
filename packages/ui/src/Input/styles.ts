import { createCSSSystemStandard } from "../theme";
import { CSSSystem } from "../styled";
import { InputProps } from "./Input";
import { InputStyleKeys } from "./";

export function Root(props: InputProps): CSSSystem {
	const { error, theme } = props;

	return createCSSSystemStandard<InputStyleKeys>({
		key: "root",
		config: theme?.Input,
		props,
		base: [
			{
				display: "flex",
				alignItems: "center",
				width: "100%",
				boxSizing: "border-box",
				borderRadius: "4px",
				border: "1px solid",
				borderColor: error ? "negative.base" : "border.base",
				backgroundColor: "#fff",
				transition: "125ms border-color ease-in",
				paddingX: "scale-2",
				paddingY: "scale-2",

				"&:focus-within": {
					borderColor: "primary.base",
				},
				"&:hover": {
					borderColor: "primary.base",
				},
			},
		],
	});
}

export function Input(props: InputProps): CSSSystem {
	const { theme } = props;
	return createCSSSystemStandard<InputStyleKeys>({
		key: "input",
		config: theme?.Input,
		props,
		base: [
			{
				padding: 0,
				width: "100%",
				height: "100%",
				border: "none",
				outline: "none",
				resize: "none",
				boxSizing: "content-box",
				background: "transparent",
				textStyle: "input",
			},
		],
	});
}
