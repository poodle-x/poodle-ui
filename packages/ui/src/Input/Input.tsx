import React from "react";
import Box, { BoxProps } from "../Box";
import { getCSSSystemBoxProps } from "../styled/CSSSystem";
import { StandardComponentProps, ThemeConfig } from "../theme";
import useDefaultProps from "../hooks/useDefaultProps/useDefaultProps";
import * as styles from "./styles";

export interface LocalInputProps {
	/**
	 * Input props
	 */
	inputProps?: BoxProps;
	/**
	 * Change the component style to error style when set to true.
	 */
	error?: boolean;
	/**
	 * Add prefix to the input.
	 */
	startAdornment?: React.ReactNode;
	/**
	 * Add suffix to the input.
	 */
	endAdornment?: React.ReactNode;
	/**
	 * Change input to textarea element.
	 */
	multiline?: boolean;
}

export interface InputProps
	extends BoxProps,
		StandardComponentProps,
		LocalInputProps {}

function getDefaultProps(theme?: ThemeConfig) {
	return theme?.Input?.defaultProps;
}

export const Input: React.ForwardRefExoticComponent<
	React.PropsWithoutRef<InputProps> & React.RefAttributes<HTMLElement>
> = React.forwardRef<HTMLElement, InputProps>((_props, ref) => {
	const { props, isLocalTheme } = useDefaultProps<InputProps>(_props, {
		themeDefaultProps: getDefaultProps,
	});

	const {
		className,
		multiline,
		startAdornment,
		endAdornment,
		inputProps,
		variant,
		error,
		...otherProps
	} = props;

	const as = multiline ? "textarea" : "input";

	return (
		<Box
			{...otherProps}
			{...getCSSSystemBoxProps({
				isRoot: true,
				isLocalTheme,
				componentProps: props,
				fnCSSSystem: styles.Root,
				baseClassName: ["poodle-input"],
			})}
			ref={ref}
		>
			{startAdornment && startAdornment}
			<Box
				as={as}
				aria-invalid={error}
				{...getCSSSystemBoxProps({
					componentProps: props,
					partProps: inputProps,
					fnCSSSystem: styles.Input,
					baseClassName: ["poodle-input_input"],
				})}
			/>
			{endAdornment && endAdornment}
		</Box>
	);
});

Input.displayName = "PoodleInput";
