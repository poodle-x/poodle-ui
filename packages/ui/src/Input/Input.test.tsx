import React from "react";
import { ThemeConfig } from "../theme";
import { render } from "../utils/test";
import Input from "./";

describe("<Input />", () => {
	it("should render Input correctly", () => {
		const { container } = render(
			<Input
				id="id"
				inputProps={{
					"aria-label": "Simple label",
				}}
			/>
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	it("should render error Input correctly", () => {
		const { container } = render(
			<Input
				error={true}
				id="id"
				inputProps={{
					"aria-label": "Simple label",
				}}
			/>
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	it("should render Input textarea correctly", () => {
		const { container } = render(
			<Input
				multiline={true}
				inputProps={{
					rows: 3,
					"aria-label": "Simple label",
				}}
			/>
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	it("should render Input with adornment correctly", () => {
		const { container } = render(
			<Input
				inputProps={{
					"aria-label": "Simple label",
				}}
				startAdornment="$"
				endAdornment="kg"
			/>
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	it("ref should work", () => {
		const ref = React.createRef<HTMLDivElement>();
		const refInput = React.createRef<HTMLDivElement>();

		render(
			<Input
				ref={ref}
				inputProps={{
					ref: refInput,
				}}
			/>
		);
		expect(ref.current).toMatchSnapshot();
		expect(refInput.current).toMatchSnapshot();
	});

	it("should render Box props correctly", () => {
		const { container } = render(
			<Input
				id="id"
				mt="20px"
				inputProps={{
					px: 10,
				}}
			/>
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	describe("theming", () => {
		const themeTest: ThemeConfig = {
			mode: "mode",
			Input: {
				styles: {
					root: {
						mt: 10,
						letterSpacing: 250,
					},
					input: {
						fontSize: 50,
					},
				},
				variants: {
					custom: {
						root: {
							top: 0,
							textIndent: 1,
						},
						input: {
							left: 100,
						},
					},
				},
			},
		};

		it("should extend styles in global", () => {
			const { container } = render(<Input variant="custom" />, {
				theme: themeTest,
			});
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should extend styles in local", () => {
			const { container } = render(
				<Input variant="custom" themeExtend={themeTest} />
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should render with default props correctly", () => {
			const { container } = render(<Input />, {
				theme: {
					Input: {
						defaultProps: {
							id: "id",
							className: "hello",
						},
					},
				},
			});
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should override styles", () => {
			const { container } = render(<Input />, {
				theme: {
					Input: {
						overrides: {
							root: {
								mt: 50,
							},
						},
					},
				},
			});
			expect(container.firstChild).toMatchSnapshot();
		});
	});
});
