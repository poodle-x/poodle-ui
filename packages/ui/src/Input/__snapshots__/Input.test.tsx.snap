// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> ref should work 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid;
  border-color: #CCCCCC;
  background-color: #fff;
  -webkit-transition: 125ms border-color ease-in;
  transition: 125ms border-color ease-in;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c0:focus-within {
  border-color: #00478A;
}

.c0:hover {
  border-color: #00478A;
}

.c1 {
  padding: 0;
  width: 100%;
  height: 100%;
  border: none;
  outline: none;
  resize: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
}

<div
  class="poodle-box poodle-input c0"
>
  <input
    class="poodle-box poodle-input_input c1"
  />
</div>
`;

exports[`<Input /> ref should work 2`] = `
.c0 {
  padding: 0;
  width: 100%;
  height: 100%;
  border: none;
  outline: none;
  resize: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
}

<input
  class="poodle-box poodle-input_input c0"
/>
`;

exports[`<Input /> should render Box props correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid;
  border-color: #CCCCCC;
  background-color: #fff;
  -webkit-transition: 125ms border-color ease-in;
  transition: 125ms border-color ease-in;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
  margin-top: 20px;
}

.c0:focus-within {
  border-color: #00478A;
}

.c0:hover {
  border-color: #00478A;
}

.c1 {
  padding: 0;
  width: 100%;
  height: 100%;
  border: none;
  outline: none;
  resize: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
  padding-left: 10px;
  padding-right: 10px;
}

<div
  class="poodle-box poodle-input c0"
  id="id"
>
  <input
    class="poodle-box poodle-input_input c1"
  />
</div>
`;

exports[`<Input /> should render Input correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid;
  border-color: #CCCCCC;
  background-color: #fff;
  -webkit-transition: 125ms border-color ease-in;
  transition: 125ms border-color ease-in;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c0:focus-within {
  border-color: #00478A;
}

.c0:hover {
  border-color: #00478A;
}

.c1 {
  padding: 0;
  width: 100%;
  height: 100%;
  border: none;
  outline: none;
  resize: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
}

<div
  class="poodle-box poodle-input c0"
  id="id"
>
  <input
    aria-label="Simple label"
    class="poodle-box poodle-input_input c1"
  />
</div>
`;

exports[`<Input /> should render Input textarea correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid;
  border-color: #CCCCCC;
  background-color: #fff;
  -webkit-transition: 125ms border-color ease-in;
  transition: 125ms border-color ease-in;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c0:focus-within {
  border-color: #00478A;
}

.c0:hover {
  border-color: #00478A;
}

.c1 {
  padding: 0;
  width: 100%;
  height: 100%;
  border: none;
  outline: none;
  resize: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
}

<div
  class="poodle-box poodle-input c0"
>
  <textarea
    aria-label="Simple label"
    class="poodle-box poodle-input_input c1"
    rows="3"
  />
</div>
`;

exports[`<Input /> should render Input with adornment correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid;
  border-color: #CCCCCC;
  background-color: #fff;
  -webkit-transition: 125ms border-color ease-in;
  transition: 125ms border-color ease-in;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c0:focus-within {
  border-color: #00478A;
}

.c0:hover {
  border-color: #00478A;
}

.c1 {
  padding: 0;
  width: 100%;
  height: 100%;
  border: none;
  outline: none;
  resize: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
}

<div
  class="poodle-box poodle-input c0"
>
  $
  <input
    aria-label="Simple label"
    class="poodle-box poodle-input_input c1"
  />
  kg
</div>
`;

exports[`<Input /> should render error Input correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid;
  border-color: #AF1308;
  background-color: #fff;
  -webkit-transition: 125ms border-color ease-in;
  transition: 125ms border-color ease-in;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c0:focus-within {
  border-color: #00478A;
}

.c0:hover {
  border-color: #00478A;
}

.c1 {
  padding: 0;
  width: 100%;
  height: 100%;
  border: none;
  outline: none;
  resize: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
}

<div
  class="poodle-box poodle-input c0"
  id="id"
>
  <input
    aria-invalid="true"
    aria-label="Simple label"
    class="poodle-box poodle-input_input c1"
  />
</div>
`;

exports[`<Input /> theming should extend styles in global 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid;
  border-color: #CCCCCC;
  background-color: #fff;
  -webkit-transition: 125ms border-color ease-in;
  transition: 125ms border-color ease-in;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
  margin-top: 10px;
  letter-spacing: 250px;
  top: 0;
  text-indent: 1px;
}

.c0:focus-within {
  border-color: #00478A;
}

.c0:hover {
  border-color: #00478A;
}

.c1 {
  padding: 0;
  width: 100%;
  height: 100%;
  border: none;
  outline: none;
  resize: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
  font-size: 50px;
  left: 100px;
}

<div
  class="poodle-box poodle-input c0"
>
  <input
    class="poodle-box poodle-input_input c1"
  />
</div>
`;

exports[`<Input /> theming should extend styles in local 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid;
  border-color: #CCCCCC;
  background-color: #fff;
  -webkit-transition: 125ms border-color ease-in;
  transition: 125ms border-color ease-in;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
  margin-top: 10px;
  letter-spacing: 250px;
  top: 0;
  text-indent: 1px;
}

.c0:focus-within {
  border-color: #00478A;
}

.c0:hover {
  border-color: #00478A;
}

.c1 {
  padding: 0;
  width: 100%;
  height: 100%;
  border: none;
  outline: none;
  resize: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
  font-size: 50px;
  left: 100px;
}

<div
  class="poodle-box poodle-input c0"
>
  <input
    class="poodle-box poodle-input_input c1"
  />
</div>
`;

exports[`<Input /> theming should override styles 1`] = `
.c0 {
  margin-top: 50px;
}

.c1 {
  padding: 0;
  width: 100%;
  height: 100%;
  border: none;
  outline: none;
  resize: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
}

<div
  class="poodle-box poodle-input c0"
>
  <input
    class="poodle-box poodle-input_input c1"
  />
</div>
`;

exports[`<Input /> theming should render with default props correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid;
  border-color: #CCCCCC;
  background-color: #fff;
  -webkit-transition: 125ms border-color ease-in;
  transition: 125ms border-color ease-in;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c0:focus-within {
  border-color: #00478A;
}

.c0:hover {
  border-color: #00478A;
}

.c1 {
  padding: 0;
  width: 100%;
  height: 100%;
  border: none;
  outline: none;
  resize: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
}

<div
  class="poodle-box poodle-input hello c0"
  id="id"
>
  <input
    class="poodle-box poodle-input_input c1"
  />
</div>
`;
